// proxypath-client/src/composed/proxypath_api.ts
import axios from "axios";
import { reactive } from "vue";
import { ListResponse } from "./model";

export const getUrl = () => {
  return "http://api.component.com:8080";
};

export const useList = () => {
  const info = reactive({
    loading: false,
    response: { data: [] } as ListResponse,
  });
  const request = (configList: string[]) => {
    info.loading = true
    if (!configList || configList.length <1) {
        return
    }
    let config = ""
    for (let i = 0; i < configList.length; i++) {
        if (i == configList.length - 1) {
            config = config + configList[i]
        } else {
            config = config + configList[i] + ","
        }
    }
    axios
      .get<ListResponse>(getUrl() + "/list", {
        params: {
            config: config
        }
      })
      .then((response) => {
        info.response = response.data
      })
      .catch((error) => {
        console.error(error)
        alert(error)
      })
      .finally(() => (info.loading = false));
  };
  return {
    info,
    request
  }
};
